File: practice/algorithms/implementation/jumping_on_the_clouds/JumpingOnTheClouds.scala
import scala.io.Source

object JumpingOnTheClouds extends App {
  val lines = Source.stdin.getLines().toList
  val clouds = lines(1).split(" ").map(_.toInt)

  var steps = 0
  var index = 0
  while (index < clouds.length - 1) {
    if((index + 2 < clouds.length) && clouds(index + 2) != 1) {
      index = index + 2
    } else {
      index = index + 1
    }
    steps += 1
  }

  println(steps)
}

File: practice/algorithms/implementation/jumping_on_the_clouds/jumping_on_the_clouds.rb
n = gets.to_i
clouds = gets.split(" ").map{|i| i.to_i}

steps = 0
index = 0
while (index < clouds.length - 1) do
    if((index + 2 < clouds.length) && clouds[index + 2] != 1)
      index = index + 2
    else
      index = index + 1
    end
    steps += 1
end

puts steps